autoblocks/_impl/testing/evaluators/llm_judge.py

```diff
@@ -0,0 +1,186 @@
+from autoblocks._impl.testing.models import ScoreChoice
+from autoblocks._impl.testing.models import TestCaseType
+from autoblocks._impl.testing.models import Threshold
+
+function_name = "select_answer"
+
+
+class BaseLLMJudge(BaseTestEvaluator, abc.ABC, Generic[TestCaseType, OutputType]):
+    """
+    Base evaluator for creating an LLM judge.
+    """
+
+    @property
+    def threshold(self) -> Optional[Threshold]:
+        """
+        The threshold for the evaluator.
+        """
+        return None
+
+    @property
+    def model(self) -> str:
+        """
+        The model to use for the evaluator.
+        It must be an OpenAI model that supports tools.
+        Defaults to "gpt-4-turbo".
+        """
+        return "gpt-4-turbo"
+
+    @property
+    def no_of_overrides(self) -> int:
+        """
+        The number of recent overrides to fetch for the evaluator and pass to make_prompt.
+
+        Defaults to 0.
+        """
+        return 0
+
+    @abc.abstractmethod
+    def make_prompt(
+        self, test_case: TestCaseType, output: OutputType, recent_overrides: List[EvaluatorOverride]
+    ) -> str:
+        """
+        The prompt passed to the LLM judge. Should be poised as a question.
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def score_choices(self) -> List[ScoreChoice]:
+        """
+        The choices for the LLM judge to use when answering.
+        """
+        pass
+
+    def _make_score_choice(self, value: float) -> Optional[ScoreChoice]:
+        return next((score for score in self.score_choices if score.value == value), None)
+
+    async def _get_recent_overrides(self) -> List[EvaluatorOverride]:
+        if self.no_of_overrides == 0:
+            # Don't fetch if the consumer hasn't requested any overrides
+            return []
+        test_id = get_test_id(evaluator_id=self.id)
+        resp = await global_state.http_client().get(
+            f"https://{API_ENDPOINT}/test-suites/{test_id}/evaluators/{self.id}/human-reviews?n={self.no_of_overrides}",
+            headers={"Authorization": f"Bearer {get_autoblocks_api_key(self.id)}"},
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        overrides = []
+        for eval_data in data:
+            override_score = self._make_score_choice(eval_data["overrideScore"])
+            original_score = self._make_score_choice(eval_data["originalScore"])
+            if override_score is None or original_score is None:
+                # If for some reason we can't find the score choice, skip this override
+                # Could happen if the score choices have changed since the override was created
+                continue
+
+            overrides.append(
+                EvaluatorOverride(
+                    original_score=original_score,
+                    override_score=override_score,
+                    output_fields=[EvaluatorOverrideField(**field) for field in eval_data["outputFields"]],
```

Comment:
i'd build this explicitly like you are for `EvaluatorOverrideComment` since right now it's incidental that none of the fields have underscores, but that might not always be the case